modules/global-catalog/src/test/java/com/ibm/cloud/platform_services/global_catalog/v1/GlobalCatalogIT.java

```diff
@@ -0,0 +1,523 @@
+
+        List<Artifact> resources = result.getResources();
+        assertNotNull(resources);
+        assertEquals(resources.size(), 1);
+        assertEquals(resources.get(0).getName(), uploadArtifact.artifactId());
+        assertEquals(resources.get(0).getUrl(), String.format("%s/%s/artifacts/%s", service.getServiceUrl(), defaultCreate.id(), uploadArtifact.artifactId()));
+        assertEquals(resources.get(0).getSize().intValue(), 23); // TODO: busted always returns a value for 4???
+    }
+
+    @Test
+    public void testListArtifactsFailure() {
+        Response<Artifacts> response = service.listArtifacts(listArtifacts).execute();
+        assertNotNull(response);
+        assertEquals(response.getStatusCode(), 200);
+
+        Artifacts result = response.getResult();
+        assertNotNull(result);
+        assertEquals(result.getCount().intValue(), 0);
+    }
+
+    @Test
+    public void testGetArtifacts() {
+        service.createCatalogEntry(defaultCreate).execute();
+        service.uploadArtifact(uploadArtifact).execute();
+        Response<Void> response = service.getArtifact(getArtifact).execute();
+        assertNotNull(response);
+        assertEquals(response.getStatusCode(), 200);
+        assertEquals(response.getResult(), uploadArtifact.artifact());     // TODO: busted, artifact not returned because of void
+    }
+
+    @Test
+    public void testGetArtifactFailure() {
+        service.createCatalogEntry(defaultCreate).execute();
+
+        try {
+            service.getArtifact(getArtifact).execute();
+            fail("Expected NotFoundException.");
+        } catch (NotFoundException e) {
+            assertEquals(e.getStatusCode(), 404);
+        }
+
+        service.deleteCatalogEntry(defaultDelete);
+
+        try {
+            service.getArtifact(getArtifact).execute();
+            fail("Expected NotFoundException.");
+        } catch (NotFoundException e) {
+            assertEquals(e.getStatusCode(), 404);
+        }
+    }
+
+    @Test
+    public void testCreateArtifact() {
+        service.createCatalogEntry(defaultCreate).execute();
+        Response<Void> response = service.uploadArtifact(uploadArtifact).execute();
+        assertNotNull(response);
+        assertEquals(response.getStatusCode(), 200);
+    }
+
+    @Test
+    public void testCreateArtifactFailure() {
+        try {
+            service.uploadArtifact(uploadArtifact).execute();
+            fail("Expected NotFoundException.");
+        } catch (NotFoundException e) {
+            assertEquals(e.getStatusCode(), 404);
+        }
+    }
+
+    @Test
+    public void testDeleteArtifact() {
+        service.createCatalogEntry(defaultCreate).execute();
+        service.uploadArtifact(uploadArtifact).execute();
+        Response<Void> response = service.deleteArtifact(deleteArtifact).execute();
+        assertNotNull(response);
+        assertEquals(response.getStatusCode(), 200);
+    }
+
+    @Test
```

Comment:
One way to do negative tests like this is to simply invoke the operation inside the test method with no try/catch, but then use the `expectedExceptions` attribute of the Test annotation, like this:
```
@Test(expectedExceptions = { NotFoundException.class })
```